### ISSUE TITLE
Can't see files in directories in password-protected zip

### ISSUE THREAD

**Issue by TheLastProject** [srikanth-lingala/zip4j#329]
Please see https://stackoverflow.com/questions/68247492/unzipping-files-in-directory-in-password-protected-zipinputstream. This seems a bug, because if the zip is not password-protected it does print "test_dir/test_file" as well.

**Comment by srikanth-lingala** [comment]
I just tried to reproduce this, and it work fine for me. [This](https://gist.github.com/srikanth-lingala/98e16f5979c20b7b91c94c0c88620411) is the code I tried. And the zip file is attached.

Can you attach your zip file that I can try with?

[sample.zip](https://github.com/srikanth-lingala/zip4j/files/6763522/sample.zip)

**Comment by TheLastProject** [comment]
Interesting, I can confirm it works fine with your sample.zip. But it fails with my test zip. I've attached my test zip.

[test_dir.zip](https://github.com/srikanth-lingala/zip4j/files/6765655/test_dir.zip)

Comparing the files, it seems like my .zip is a newer version:
```
$ file sample.zip
sample.zip: Zip archive data, at least v2.0 to extract
$ file test_dir.zip
test_dir.zip: Zip archive data, at least v5.1 to extract
```

Debugging further with your very useful example, it turned out my original issue (with a v1.0 archive) was just me not realizing zip4j (logically) reading to the end of the input stream and that I had to get a new input stream. Oops...

But anyway, it does seem that there are some actual issues with password-protected v5.1+ .zip files.

Putting your and my example zip side-by-side, it seems that in my case my zip tooling password-protected both the directory and the file in it, while yours didn't:
![image](https://user-images.githubusercontent.com/1885159/124504177-39a0c500-ddc7-11eb-91fc-fe0e994e4388.png)

**Comment by srikanth-lingala** [comment]
> it does seem that there are some actual issues with password-protected v5.1+ .zip files.

This is an interesting scenario. Usually, zip tools don't encrypt a directory. There is no advantage in encrypting a directory name. You can still read the name of the directory without a password, and there is no content in a directory (in a zip format). It is just an entry with the name of the directory. So not a lot of tools encrypt directories. In fact, this is the first time I came across this behaviour. Which tool did you use to create the zip file?

But, I agree that this is a bug in zip4j. When a directory is encrypted, zip4j does not read the entries after that when reading with Inputstreams. I will fix this, hopefully in the next release.

**Comment by TheLastProject** [comment]
I created this zip with [Ark](https://apps.kde.org/ark/), the default zip tool for the Linux desktop environment KDE.

I used the following settings:
![image](https://user-images.githubusercontent.com/1885159/124507066-4b856680-ddcd-11eb-92a1-80e07193be74.png)

**Comment by srikanth-lingala** [comment]
I tried with your test_dir.zip and it seems that zip4j now prints both entries and doesn't just stop after printing the first entry. This issue might be fixed by some other fix which was done in the meantime. Marking this issue as resolved. This is the output I got:

```
test_dir/
test_dir/test_file
```

Please note that the version I tried with is a SNAPSHOT which is not released yet. It might be that the fix is in this snapshot version. This snapshot version will be released in a couple of days.

**Comment by srikanth-lingala** [comment]
Fixed in v2.10.0 released today
